paddle/fluid/framework/ir/graph_pattern_detector.h

```diff
@@ -1032,6 +1032,22 @@ struct Elementwise : public PatternBase {
   PATTERN_DECL_NODE(elementwise_out);
 };
 
+// Residual Elementwise ops
+// This pattern allows operator (FC/Conv) output to be X or Y
+// and residual data Y or X, based on flag
+struct ResidualElementwise : public PatternBase {
+  ResidualElementwise(PDPattern* pattern, const std::string& name_scope,
+                      bool asX)
+      : PatternBase(pattern, name_scope, "residual_elementwise") {}
+  PDNode* operator()(PDNode* op_var, PDNode* residual_var,
+                     const std::string elementwise_type, bool asX);
```

Comment:
It's not a big deal, but could you please change `asX` to `as_x` to preserve using lower snake case?